### ISSUE TITLE
Ex: Add custom field from Companies / People table options menu

### ISSUE THREAD

**Issue by Parvezkhan0** [twentyhq/twenty#2073]
## Scope & Context

 Implementing a User Notifications feature to enhance user engagement and productivity. 
## Current behavior

Currently, the system lacks a user notification feature. Users are not receiving any notifications for activities within the CRM. There is no UI for notifications, and no data is being stored in the database for this purpose.

## Expected behavior
Backend Development:

- Implement a notification system that tracks and sends notifications to users in real-time.

- Notifications should be triggered by activities such as new messages, task assignments, or opportunity updates.

- Users should be able to customize their notification preferences in their user profile.

Frontend Development:

- Design and integrate a user-friendly notification UI that displays real-time notifications.
- Users should receive notifications as pop-up alerts within the CRM interface.
- Notifications should be visually distinguishable based on their type (e.g., messages, tasks, opportunities).
- Users should have the option to clear notifications individually or all at once.

Database Schema:

- Create a schema for storing notifications in the database.

- Each notification entry should contain information about the type of notification, sender, recipient, timestamp, and relevant content.

- Implement database queries to store and retrieve notifications efficiently.

## Technical inputs

Backend Development:

- Create a notification service responsible for tracking and sending notifications.

- Implement the logic to trigger notifications based on CRM activities.

- Ensure that notification preferences are stored in the user's profile.

Frontend Development:

- Design a notification pop-up component that displays real-time notifications.

- Integrate the notification UI into the CRM interface.

- Ensure that notifications are visually distinguishable based on their type.

- Provide options for clearing notifications.

Database Schema:

- Define the database schema for storing notifications.

- Create tables and fields to store notification data.

- Implement database queries for efficient notification storage and retrieval.

**Comment by brendanlaschke** [comment]
Thats a great idea - its already on the roadmap ;), with design and some thoughts how to implement it. 
See #815 for the details ...

**Comment by charlesBochet** [comment]
Yes! This is likely something we will do before end of the year. 
Firstly we need to implement a queue/job mechanism to properly process notifications or send email. Doing it synchronously would not be good endpoint response time.
We will likely implement queues in October, then we can start planning for notifications.

Thanks @brendanlaschke, always up-to-date ;)

**Comment by charlesBochet** [comment]
I'm converting this issue to discussion
